platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -297,6 +304,71 @@ @implementation NSExpression (MGLAdditions)
+    NSPredicate *conditional = [NSPredicate predicateWithFormat:conditionPredicate];
+    return [NSExpression expressionForConditional:conditional trueExpression:trueExpression falseExpression:falseExpression];
+}
+
++ (instancetype)mgl_expressionForStepFunction:(MGLExpressionStyleFunction)function defaultValue:(NSValue *)value stops:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)stops {
+    return [NSExpression mgl_expressionForStepFunction:[NSExpression expressionForVariable:function]
+                                     defaultExpression:[NSExpression mgl_expressionForValue:value]
+                                                 stops:[NSExpression expressionWithFormat:@"%@", stops]];
+}
+
++ (instancetype)mgl_expressionForStepFunction:(MGLExpressionStyleFunction)function defaultString:(NSString *)string stops:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)stops {
+    return [NSExpression mgl_expressionForStepFunction:[NSExpression expressionForVariable:function]
+                                     defaultExpression:[NSExpression mgl_expressionForString:string]
+                                                 stops:[NSExpression expressionWithFormat:@"%@", stops]];
+}
+
++ (instancetype)mgl_expressionForStepFunction:(MGLExpressionStyleFunction)function defaultColor:(MGLColor *)color stops:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)stops {
+    return [NSExpression mgl_expressionForStepFunction:[NSExpression expressionForVariable:function]
+                                     defaultExpression:[NSExpression mgl_expressionForColor:color]
+                                                 stops:[NSExpression expressionWithFormat:@"%@", stops]];
+}
+
++ (instancetype)mgl_expressionForStepFunction:(NSExpression *)operatorExpression defaultExpression:(NSExpression *)expression stops:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)stops {
+    return [NSExpression expressionForFunction:operatorExpression selectorName:@"mgl_stepWithMinimum:stops:" arguments:@[expression, stops]];
+}
+
++ (instancetype)mgl_expressionForInterpolateFunction:(MGLExpressionStyleFunction)function curveType:(nonnull MGLExpressionInterpolationMode)curveType steps:(nonnull NS_DICTIONARY_OF(NSNumber *, id) *)steps; {
+    return [NSExpression mgl_expressionForInterpolateFunction:[NSExpression expressionForVariable:function]
+                                                    curveType:curveType
+                                                   parameters:nil
+                                                        steps:[NSExpression expressionWithFormat:@"%@", steps]];
+}
+
++ (instancetype)mgl_expressionForInterpolateFunction:(nonnull NSExpression*)expressionOperator curveType:(nonnull MGLExpressionInterpolationMode)curveType parameters:(nullable NSExpression *)parameters steps:(nonnull NSExpression*)steps {
+    NSExpression *sanitizeParams = parameters ? parameters : [NSExpression expressionWithFormat:@"%@", parameters];
```

Comment:
If `parameters` is `nil`, then `sanitizeParams` becomes a constant value expression representing `nil`. Consider writing that out explicitly:

```objc
parameters = parameters ?: [NSExpression expressionForConstantValue:nil];
```